errorcode_book_generator/src/main.rs

```diff
@@ -0,0 +1,138 @@
+
+    if supported {
+        std::process::exit(0);
+    } else {
+        eprintln!("{renderer} is not supported");
+        std::process::exit(1);
+    }
+}
+
+struct ErrorCodeGenerator {}
+
+impl Preprocessor for ErrorCodeGenerator {
+    fn name(&self) -> &str {
+        "errors"
+    }
+
+    fn run(
+        &self,
+        ctx: &mdbook::preprocess::PreprocessorContext,
+        mut book: mdbook::book::Book,
+    ) -> mdbook::errors::Result<mdbook::book::Book> {
+        //Get a config location on where to read the errors from
+        //Get a config location on where to place the error chapters in the book
+        let preprocessor = ctx.config.get_preprocessor(self.name()).expect("Preprocessor configured");
+        let Some((source, target)) = preprocessor
+            .get("err_location")
+            .and_then(Value::as_str)
+            .zip(preprocessor.get("err_target").and_then(Value::as_str))
+        else {
+            return Err(Error::msg(
+                "Invalid configuration, the keys \"err_location\" and \"err_target\" must be set",
+            ));
+        };
+
+        //Read the config source for all md files
+        let source = ctx.root.join(PathBuf::from_str(source)?);
+        let target = PathBuf::from_str(target)?;
+        if !source.exists() {
+            return Err(Error::msg(format!("{}: Invalid Path", source.to_string_lossy())));
+        }
+
+        //Find the section in the book where to add the chapters
+        let mut files: Vec<_> = fs::read_dir(source)?
+            .flatten()
+            .map(|it| it.path())
+            .filter(|it| it.extension().unwrap_or_default() == "md")
+            .collect();
+        files.sort();
+        book.for_each_mut(|item| {
+            if let BookItem::Chapter(chapter) = item {
+                if chapter.source_path == Some(target.clone()) {
+                    let mut parents = chapter.parent_names.clone();
+                    parents.push(chapter.name.clone());
+                    let mut number = 1;
+                    for file in &files {
+                        //Add each file to the book
+                        let name = file.file_stem().and_then(|it| it.to_str()).unwrap_or_default();
+                        let content = std::fs::read_to_string(file).unwrap();
+                        let mut error_chapter = Chapter::new(
+                            name,
+                            content,
+                            &target.join(name).with_extension("md"),
+                            parents.clone(),
+                        );
+                        let mut chapter_number = chapter.number.clone().unwrap_or_default();
+                        chapter_number.push(number);
+                        error_chapter.number = Some(chapter_number);
+                        chapter.sub_items.push(error_chapter.into());
+                        number += 1;
+                    }
+                }
+            }
+        });
```

Comment:
Do you think this snippet could benefit from a short high-level comment as to what exactly happens here? Or perhaps a README file in the `errorcode_book_generator/` which a high level documentation?

Reply:
Done